src/Orchard.Web/Modules/Orchard.Redis/Caching/RedisCacheStorageProvider.cs

```diff
@@ -10,9 +10,13 @@
 
 namespace Orchard.Redis.Caching {
 
+    public interface IRedisCacheStorageProvider : ICacheStorageProvider {
+        void Clear(string key);
+    }
+
     [OrchardFeature("Orchard.Redis.Caching")]
     [OrchardSuppressDependency("Orchard.Caching.Services.DefaultCacheStorageProvider")]
-    public class RedisCacheStorageProvider : Component, ICacheStorageProvider {
+    public class RedisCacheStorageProvider : Component, IRedisCacheStorageProvider {
```

Comment:
Will this resolve the service on both interfaces?

Reply:
Yes, it does. I use this pattern very often in Orchard and never had any issues with it.